src/main/java/com/fasterxml/jackson/databind/introspect/JacksonAnnotationIntrospector.java

```diff
@@ -240,6 +240,45 @@ public String[] findEnumValues(Class<?> enumType, Enum<?>[] enumValues, String[]
         return names;
     }
 
+    @Override
+    public String[] findEnumValues(MapperConfig<?> config, Enum<?>[] enumValues, String[] names,
+                                   AnnotatedClass annotatedClass) {
+        // First collect all JsonProperty.value()
+        HashMap<String, String> enumToPropertyMap = new HashMap<>();
+        for (AnnotatedField field : annotatedClass.fields()) {
+            JsonProperty property = field.getAnnotation(JsonProperty.class);
+            if (property != null) {
+                enumToPropertyMap.put(field.getName(), property.value());
```

Comment:
Should this check for empty name ("")? If existing code doesn't check, that's fine I guess but I think often happens when just adding `@JsonProperty` as-is (useless annotation for Enums of course), or specifying other properties but not `value`.

Reply:
> Done 👍🏻 Revised(moved) this check to right before adding it to the Map.

Done, in accordance with [your other comment](https://github.com/FasterXML/jackson-databind/pull/3832/files#r1228798773) 👍🏻